src/lib/card.ts

```diff
@@ -0,0 +1,200 @@
+import {roam} from "./core";
+import {InfoHelper} from "../index";
+import {setCookie, uFetch} from "../utils/network";
+import {
+    CARD_CANCEL_LOSS_URL, CARD_CHANGE_PWD_URL,
+    CARD_INFO_BY_USER_URL,
+    CARD_LOGIN_URL, CARD_MOD_MAX_CONSUME_URL,
+    CARD_PHOTO_URL, CARD_RECHARGE_FROM_BANK_URL, CARD_RECHARGE_FROM_WECHAT_ALIPAY_URL, CARD_REPORT_LOSS_URL,
+    CARD_TRANSACTION_URL, CARD_USER_BY_TOKEN_URL,
+    CONTENT_TYPE_JSON,
+} from "../constants/strings";
+import {CardInfo} from "../models/card/info";
+import {CardTransaction, CardTransactionType} from "../models/card/transaction";
+
+const accountBaseInfo = {
+    user: "",
+    cardId: "",
+};
+
+const parseResultData : any = (data: string) => {
+    const result = JSON.parse(data);
+    if (result.success !== true) {
+        throw new Error(result.message);
+    }
+    return result.resultData;
+};
+
+const fetchWithParse = async (url: string, jsonStruct: any = {}) => {
+    const response = await uFetch(url, JSON.stringify(jsonStruct) as any, undefined, undefined, true, CONTENT_TYPE_JSON);
+    return parseResultData(response);
+};
+
+export const cardLogin = async (helper: InfoHelper): Promise<string> => {
+    const redirectUrl = await roam(helper, "card", "eea30cbedcaf97c69d28b2d92f22a259");
+    const ticket = new RegExp(/ticket=(\w*?)$/).exec(redirectUrl)![1];
+    const token = (await fetchWithParse(CARD_LOGIN_URL, {ticket: ticket})).token;
+    setCookie("token", token);
+    accountBaseInfo.user = parseResultData(await uFetch(CARD_USER_BY_TOKEN_URL)).loginuser;
+    return token;
+};
+
+export const cardGetInfo = async (helper: InfoHelper): Promise<CardInfo> => {
+    if (accountBaseInfo.user === "") {
```

Comment:
It is not a good practice to test the login status in this way, as the token may expire and the APP may live longer than the token.

Reply:
Fine. Will change that later.